Buildings/Fluid/CHPs/BaseClasses/AssertPower.mo

```diff
@@ -0,0 +1,127 @@
+within Buildings.Fluid.CHPs.BaseClasses;
+model AssertPower "Assert if electric power is outside boundaries"
+  extends Modelica.Blocks.Icons.Block;
+
+  replaceable parameter Buildings.Fluid.CHPs.Data.Generic per
+    "Performance data"
+    annotation (Placement(transformation(extent={{-80,60},{-60,80}})));
+
+  Buildings.Controls.OBC.CDL.Interfaces.RealInput PEleDem(
+    final unit="W")
+    "Electric power demand"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}}),
+      iconTransformation(extent={{-140,-20},{-100,20}})));
+  Buildings.Controls.OBC.CDL.Utilities.Assert assMesP(
+    final message="Electric power is outside boundaries!")
+    "Generate warning when the electric power demand is out of the range"
+    annotation (Placement(transformation(extent={{80,10},{100,30}})));
+  Buildings.Controls.OBC.CDL.Utilities.Assert assMesDP(
+    final message="Power rate of change is outside boundaries!")
+    "Assert function for checking power rate"
+    annotation (Placement(transformation(extent={{80,-50},{100,-30}})));
+
+protected
+  Buildings.Controls.OBC.CDL.Logical.Nor nor
+    "Check if the electric power demand is out of the power production range"
+    annotation (Placement(transformation(extent={{40,10},{60,30}})));
+  Buildings.Controls.OBC.CDL.Logical.Nand nand "Logical Nand"
+    annotation (Placement(transformation(extent={{40,-50},{60,-30}})));
+  Buildings.Controls.OBC.CDL.Logical.Sources.Constant cheDPLim(
+    final k=per.dPEleLim)
+    "Check if dP is limited"
+    annotation (Placement(transformation(extent={{0,-90},{20,-70}})));
+  Buildings.Controls.OBC.CDL.Continuous.Derivative demRat(
+    final initType=Buildings.Controls.OBC.CDL.Types.Init.InitialState,
+    final x_start=0)
+    "Power demand rate"
+    annotation (Placement(transformation(extent={{-80,-50},{-60,-30}})));
+  Buildings.Controls.OBC.CDL.Continuous.Abs abs1 "Absolute value"
+    annotation (Placement(transformation(extent={{-40,-50},{-20,-30}})));
+  Buildings.Controls.OBC.CDL.Continuous.GreaterEqualThreshold maxRat(
+    final threshold=per.dPEleMax)
+    "Check if demand rate is more than the maximum rate"
+    annotation (Placement(transformation(extent={{0,-50},{20,-30}})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis maxPow(
+    final uLow=0.99*per.PEleMax - 2e-6,
+    final uHigh=1.01*per.PEleMax - 1e-6)
+    "Check if the electric power demand is more than the maximum power production"
+    annotation (Placement(transformation(extent={{-40,30},{-20,50}})));
+  Buildings.Controls.OBC.CDL.Continuous.Hysteresis minPow(
+    final uLow=0.99*per.PEleMin - 2e-6,
+    final uHigh=1.01*per.PEleMin - 1e-6)
```

Comment:
In case when `per.PEleMin` is 0, then `uLow=uHigh`, which will cause error. Adding the small number 1e-6 would avoid that happen.

Reply:
Thanks, I missed that. 
Is there a reason why using different offset values though, instead of 
```
uLow=0.99*per.PEleMin - 1e-6,
uHigh=1.01*per.PEleMin + 1e-6
```
Or even
```
uLow=per.PEleMin,
uHigh=1.01*per.PEleMin + 1e-6
```

Reply:
@JayHuLBL to address all occurrences of Hysteresis